js/piwik.js

```diff
@@ -592,6 +592,43 @@ if (typeof window.Matomo !== 'object') {
 
             return matches ? matches[1] : url;
         }
+        function isPositiveNumberString(str) {
+            // !isNaN(str) could be used but does not cover '03' (octal) and '0xA' (hex)
+            // nor negative numbers
+            return (/^[1-9][0-9]*(\.[0-9]+)?$/).test(str);
+        }
+        function onlyNumbers(data) {
+            var result = {}, k;
+            for (k in data) {
+                if (data.hasOwnProperty(k)) {
+                    if (isNumber(data[k]) || isPositiveNumberString(data[k])) {
+                        result[k] = data[k];
+                    } else if (isDefined(data[k])) {
+                        throw new Error('Parameter "' + k + '" provided value "' + data[k] +
+                            '" is not valid. Please provide a numeric value.');
+                    }
+                }
+            }
+            return result;
+        }
+        function roundNumbers(data) {
+            var result = {}, k;
+            for (k in data) {
+                if (data.hasOwnProperty(k)) {
+                    result[k] = Math.round(Number(data[k]));
```

Comment:
just fyi the `Number()` call wouldn't be needed here. No need to change it though.

Reply:
I think it'll be a good change and reduces the amount of code. I didn't know that although it seems obvious now since js is loosely typed.